### ISSUE TITLE
Missing binary installer for arm64 architecture

### ISSUE THREAD

**Issue by simez** [serverless/serverless#8696]
Attempting installation via `curl -o- -L https://slss.io/install | bash` on an M1 mac results in the following error:

Sorry, there's no serverless binary installer available for arm64 architecture. Please open request for it at https://github.com/serverless/serverless/issues

This is a fresh install on a fairly new machine, most of template bits for submitting a new issue were not applicable to this problem, I hope this format is acceptable. I'm happy to help out with any debugging or providing any resources required to help support the new macs.

**Comment by pgrzesik** [comment]
Hello @simez, thanks for reporting your issue here. 

To be honest, I'm not an expert when it comes to M1 SoC compatibility/support, quick research shows me that it might be problematic as `pkg` has issues supporting `arm64`/`aarch64` out of the box (https://github.com/vercel/pkg/issues/361) and native builds of `node` for M1 Mac are available only for `v15` which is not an LTS: https://github.com/nodejs/build/issues/2474 

The current installer fails due to architecture check mismatch, but I believe the binary might work fine via Rosetta 2 - do you think that would be an acceptable solution for now, while we explore potential options to provide a native binary?

**Comment by g-balas** [comment]
Installer working fine with terminal open with [Rosetta](https://osxdaily.com/2020/11/18/how-run-homebrew-x86-terminal-apple-silicon-mac/); `sls` is then working as expected, even with native Terminal.

**Comment by simez** [comment]
Good interim measure here, looks like two core dependencies need to be updated before the ARM support can be done properly here. Please let me know if there's anything I can do to expedite this effort.

**Comment by amine2233** [comment]
Install serverless using the new homebrew version for Mac m1 `brew install serverless` and you will install the version for arm64_big_sur

```bash
Downloading https://homebrew.bintray.com/bottles/serverless-2.29.0.arm64_big_sur.bottle.tar.gz
```

**Comment by rob-gonz** [comment]
ChromeOS here, m1 Macbook fix is great but won't work for a Chromebook.  I know, "what are you doing developing on a Chromebook?".  It makes it super easy to dabble in various things while on the go.

**Comment by pgrzesik** [comment]
Hello @rob-gonz - I don't have much experience with Chromebook, but do you have a chance to install Node and `serverless` directly via `npm`?

**Comment by rob-gonz** [comment]
Hey @pgrzesik , Yes, that's what I ended up doing and it installed just fine. 

Seeing "Install as standalone binary" as the first option on https://www.serverless.com/framework/docs/getting-started/ sent me down the wrong path initially.  I then realized that's not how I have it installed on the other machines I use it on. I finally installed it from the instructions on main Github page.

Thank you for reaching out.

**Comment by marcelloromani** [comment]
I believe the "getting started" page should mention `brew install serverless`.

**Comment by mark-brooks-roostify** [comment]
A serverless binary installer for arm64 would be good, lots of people using M1 Macs now for development.

**Comment by simlu** [comment]
Is this still a problem?

**Comment by pgrzesik** [comment]
Hello, starting with next Framework release it will be possible to install binary version of `serverless`. It's still built for `x64` so it needs Rosetta to run, but the standalone installation script will now permit such installation.

**Comment by mnapoli** [comment]
@pgrzesik should we close this issue?

**Comment by pgrzesik** [comment]
Hey @mnapoli - sorry for not answering sooner. I don't think we should close this issue as the current approach only addresses it partially - it's possible to install binary on M1 Macs, but it's still running via Rosetta and isn't built natively for arm64 arch. I think if we were to close this issue, we should open another one that will be used for tracking support for native M1 binary specifically. What do you think?

**Comment by mnapoli** [comment]
👍  no problem I wasn't sure, do what you prefer (keeping this one or opening a new one)

**Comment by pgrzesik** [comment]
Thanks @mnapoli, I've edited the task title a bit and added explanation in original description to reflect the current state.

**Comment by himanshumps** [comment]
is there any update on this?

**Comment by iancward** [comment]
It should be noted that mac is not the only possible use case for this. 
There may be some people who prefer to use ARM-based Linux workstations, such as Chromebook (mentioned earlier in this issue) or Raspberry Pi. Also, Windows on ARM is also a thing and WSL (Windows Subsystem for Linux--something akin to a Linux VM) doesn't have something like Rosetta to handle arm-amd translation.

For me specifically, my team builds a general purpose docker image for our CI/CD pipelines, and some of the targets are linux arm. We're currently unable to bundle serverless in the image for arm because there isn't a standalone binary install for that platform. We could install NodeJS and install it that way, but we try to keep components small, because we put a lot of other things into the image.

**Comment by marcelloromani** [comment]
In addtion to @iancward 's comment, AWS is pushing Graviton (ARM-based) instances quite heavily.

**Comment by shearn89** [comment]
Yep, just ran into this trying to compile to deploy a Lambda to Graviton to investigate cost/speed improvements. Looks like I will need to install Node on the CI runner to get this to work?
